pxtblocks/fields/field_note.ts

```diff
@@ -74,260 +74,156 @@ namespace pxtblockly {
+        44: { name: lf("E"), prefixedName: lf("Middle E"), freq: 330 },
+        45: { name: lf("F"), prefixedName: lf("Middle F"), freq: 349 },
+        46: { name: lf("F#"), prefixedName: lf("Middle F#"), freq: 370 },
+        47: { name: lf("G"), prefixedName: lf("Middle G"), freq: 392 },
+        48: { name: lf("G#"), prefixedName: lf("Middle G#"), freq: 415 },
+        49: { name: lf("A"), prefixedName: lf("Middle A"), freq: 440 },
+        50: { name: lf("A#"), prefixedName: lf("Middle A#"), freq: 466 },
+        51: { name: lf("B"), prefixedName: lf("Middle B"), freq: 494 },
+
+        52: { name: lf("C"), prefixedName: lf("Tenor C"), altPrefixedName: lf("High C"), freq: 523 },
+        53: { name: lf("C#"), prefixedName: lf("Tenor C#"), altPrefixedName: lf("High C#"), freq: 554 },
+        54: { name: lf("D"), prefixedName: lf("Tenor D"), altPrefixedName: lf("High D"), freq: 587 },
+        55: { name: lf("D#"), prefixedName: lf("Tenor D#"), altPrefixedName: lf("High D#"), freq: 622 },
+        56: { name: lf("E"), prefixedName: lf("Tenor E"), altPrefixedName: lf("High E"), freq: 659 },
+        57: { name: lf("F"), prefixedName: lf("Tenor F"), altPrefixedName: lf("High F"), freq: 698 },
+        58: { name: lf("F#"), prefixedName: lf("Tenor F#"), altPrefixedName: lf("High F#"), freq: 740 },
+        59: { name: lf("G"), prefixedName: lf("Tenor G"), altPrefixedName: lf("High G"), freq: 784 },
+        60: { name: lf("G#"), prefixedName: lf("Tenor G#"), altPrefixedName: lf("High G#"), freq: 831 },
+        61: { name: lf("A"), prefixedName: lf("Tenor A"), altPrefixedName: lf("High A"), freq: 880 },
+        62: { name: lf("A#"), prefixedName: lf("Tenor A#"), altPrefixedName: lf("High A#"), freq: 932 },
+        63: { name: lf("B"), prefixedName: lf("Tenor B"), altPrefixedName: lf("High B"), freq: 988 },
+
+        64: { name: lf("C"), prefixedName: lf("High C"), freq: 1046 },
+        65: { name: lf("C#"), prefixedName: lf("High C#"), freq: 1109 },
+        66: { name: lf("D"), prefixedName: lf("High D"), freq: 1175 },
+        67: { name: lf("D#"), prefixedName: lf("High D#"), freq: 1245 },
+        68: { name: lf("E"), prefixedName: lf("High E"), freq: 1319 },
+        69: { name: lf("F"), prefixedName: lf("High F"), freq: 1397 },
+        70: { name: lf("F#"), prefixedName: lf("High F#"), freq: 1478 },
+        71: { name: lf("G"), prefixedName: lf("High G"), freq: 1568 },
+        72: { name: lf("G#"), prefixedName: lf("High G#"), freq: 1661 },
+        73: { name: lf("A"), prefixedName: lf("High A"), freq: 1760 },
+        74: { name: lf("A#"), prefixedName: lf("High A#"), freq: 1865 },
+        75: { name: lf("B"), prefixedName: lf("High B"), freq: 1976 }
     }
 
-    let regex: RegExp = /^Note\.(.+)$/;
-
     export interface FieldNoteOptions extends Blockly.FieldCustomOptions {
         editorColour?: string;
         minNote?: string;
         maxNote?: string;
+        eps?: string;
```

Comment:
what is eps?

Reply:
eps is the 'give' for converting input values into named notes - e.g. Middle D is 294, but by default if you type in / decompile a value between 292.1 and 295.9 (eps of 2) it will switch it to middle d. Can remove from field note options if it doesn't seem useful, it just looked like it was intended to be configurable at some point as it was the only public (non-blockly required) field [here](https://github.com/microsoft/pxt/blob/master/pxtblocks/fields/field_note.ts#L163) from 3 years ago